lib/buy/order.dart

```diff
@@ -42,24 +42,24 @@ class Order extends HiveObject {
   @HiveField(3)
   String? to;
 
-  @HiveField(4)
+  @HiveField(4, defaultValue: '')
   late String stateRaw;
 
   TradeState get state => TradeState.deserialize(raw: stateRaw);
 
   @HiveField(5)
   DateTime createdAt;
 
-  @HiveField(6)
+  @HiveField(6, defaultValue: '')
   String amount;
 
-  @HiveField(7)
+  @HiveField(7, defaultValue: '')
   String receiveAddress;
 
-  @HiveField(8)
+  @HiveField(8, defaultValue: '')
   String walletId;
 
-  @HiveField(9)
+  @HiveField(9, defaultValue: 0)
   late int providerRaw;
```

Comment:
with the current implementation of the constructor, this variable may not be initialized if the provider is null, resulting in a `late variable not initialized` exception